assets/js/sections/notes.js

```diff
@@ -1,68 +1,487 @@
+                } else if (sortState.direction === 'asc') {
+                    sortState.direction = 'desc';
+                } else if (sortState.direction === 'desc') {
+                    sortState.direction = null;
+                    sortState.column = null;
+                } else {
+                    sortState.direction = 'asc';
+                }
+                updateSortIndicators();
+                performNotesSearch();
+            });
+        } else {
+            th.style.cursor = 'default';
+        }
+    });
+
+    function updateSortIndicators() {
+        columnHeaders.forEach(function(th, idx) {
+            var span = th.querySelector('.dt-column-order');
+            if (!span) return;
+            span.textContent = '';
+            if (sortState.column === idx) {
+                if (sortState.direction === 'asc') {
+                    span.textContent = '▲';
+                } else if (sortState.direction === 'desc') {
+                    span.textContent = '▼';
+                }
+            }
+        });
+    }
+
+    // Server-side pagination, sorting, and search integration
+    var notesPerPage = 15;
+    var currentPage = 1;
+    var totalPages = 1;
+    var lastResponseTotal = 0;
+    window.lastNotesData = [];
+    var paginationContainer = document.getElementById('notesPagination');
+    if (!paginationContainer) {
+        paginationContainer = document.createElement('div');
+        paginationContainer.id = 'notesPagination';
+        paginationContainer.className = 'd-flex justify-content-center my-3';
+        var notesTableContainer = document.getElementById('notesTableContainer');
+        if (notesTableContainer) {
+            notesTableContainer.appendChild(paginationContainer);
+        }
+    }
+
+    function renderPagination() {
+        paginationContainer.innerHTML = '';
+        if (totalPages <= 1) return;
+        var ul = document.createElement('ul');
+        ul.className = 'pagination pagination-sm';
+        for (var i = 1; i <= totalPages; i++) {
+            var li = document.createElement('li');
+            li.className = 'page-item' + (i === currentPage ? ' active' : '');
+            var a = document.createElement('a');
+            a.className = 'page-link';
+            a.href = '#';
+            a.textContent = i;
+            a.addEventListener('click', function(e) {
+                e.preventDefault();
+                var page = parseInt(this.textContent);
+                if (page !== currentPage) {
+                    currentPage = page;
+                    performNotesSearch();
+                }
+            });
+            li.appendChild(a);
+            ul.appendChild(li);
+        }
+        paginationContainer.appendChild(ul);
+    }
+
+    function renderNotesTable(data) {
+        var tbody = '';
+        if (data.length === 0) {
+            tbody = '<tr><td colspan="5" class="text-center text-muted">No notes were found.</td></tr>';
```

Comment:
"No notes were found" can be also translated. We have a little go around here since our  Gettex Implementation does not support Javascript. We place generic translations in the head of the footer (here: https://github.com/wavelog/wavelog/blob/7261290151912b25bc385b874f73c0a0d580be35/application/views/interface_assets/footer.php#L29-L58) and the simply use the javascript variable here. 


```suggestion
            tbody = '<tr><td colspan="5" class="text-center text-muted">' + lang_no_notes_found + '</td></tr>';
```

Add this in the mentiond footer section:

```javascript
var lang_no_notes_found = "<?= __("No notes were found"); ?>";
```